core/state.h

```diff
@@ -19,415 +19,415 @@ limitations under the License.
-            std::abort();
-            return "";  // Quiet, compiler.
-        }
+        return t & 0x10;
     }
-
-    /** Convert the value's type into a string, for error messages. */
-    std::string type_str(const Value &v)
-    {
-        return type_str(v.t);
+};
+
+/** Convert the type into a string, for error messages. */
+std::string type_str(Value::Type t)
+{
+    switch (t) {
+        case Value::NULL_TYPE: return "null";
+        case Value::BOOLEAN: return "boolean";
+        case Value::DOUBLE: return "double";
+        case Value::ARRAY: return "array";
+        case Value::FUNCTION: return "function";
+        case Value::OBJECT: return "object";
+        case Value::STRING: return "string";
+        default:
+        std::cerr << "INTERNAL ERROR: Unknown type: " << t << std::endl;
+        std::abort();
+        return "";  // Quiet, compiler.
     }
+}
 
-    struct HeapThunk;
-
-    /** Stores the values bound to variables.
-     *
-     * Each nested local statement, function call, and field access has its own binding frame to
-     * give the values for the local variable, function parameters, or upValues.
-     */
-    typedef std::map<const Identifier*, HeapThunk*> BindingFrame;
+/** Convert the value's type into a string, for error messages. */
+std::string type_str(const Value &v)
+{
+    return type_str(v.t);
+}
 
-    /** Supertype of all objects.  Types of Value::OBJECT will point at these.  */
-    struct HeapObject : public HeapEntity {
-    };
+struct HeapThunk;
 
-    /** Hold an unevaluated expression.  This implements lazy semantics.
-     */
-    struct HeapThunk : public HeapEntity {
-        /** Whether or not the thunk was forced. */
-        bool filled;
+/** Stores the values bound to variables.
+ *
+ * Each nested local statement, function call, and field access has its own binding frame to
+ * give the values for the local variable, function parameters, or upValues.
+ */
+typedef std::map<const Identifier*, HeapThunk*> BindingFrame;
 
-        /** The result when the thunk was forced, if filled == true. */
-        Value content;
+/** Supertype of all objects.  Types of Value::OBJECT will point at these.  */
+struct HeapObject : public HeapEntity {
+};
 
-        /** Used in error tracebacks. */
-        const Identifier *name;
+/** Hold an unevaluated expression.  This implements lazy semantics.
+ */
+struct HeapThunk : public HeapEntity {
+    /** Whether or not the thunk was forced. */
+    bool filled;
 
-        /** The captured environment.
-         *
-         * Note, this is non-const because we have to add cyclic references to it.
-         */
-        BindingFrame upValues;
+    /** The result when the thunk was forced, if filled == true. */
+    Value content;
 
-        /** The captured self variable, or nullptr if there was none.  \see CallFrame. */
-        HeapObject *self;
+    /** Used in error tracebacks. */
```

Comment:
Yes, looks that way. This patch is purely whitespace changes only aside from the additions of the `// namespace` comments to the closing braces for the namespaces.